Framework/Kernel/src/NexusHDF5Descriptor.cpp

```diff
@@ -97,19 +97,18 @@ herr_t readStringAttribute(hid_t attr, char **data) {
  * Reads a string attribute of N-dimensions
  * @param attr input HDF5 attribute handler
  * @param data output attribute data
- * @param maxlen
  * @return
  */
-herr_t readStringAttributeN(hid_t attr, char *data, int maxlen) {
+std::pair<std::string, herr_t> readStringAttributeN(hid_t attr, char *data) {
```

Comment:
Do we need `data` ta all now? `dataString` is assigned to completely anyway? Can we be more direct and :
```c++
std::pair<std::string, herr_t> readStringAttributeN(hid_t attr) {
  char *vdat = NULL;
  const auto iRet = readStringAttribute(attr, vdat);
  std::string attrData;
  if(iRet >= 0) {
    attrData = vDat;
  }
  return std::make_pair(attrData, iRet);
}
```

Reply:
If `iRet is not >=0` then `dataString != vdat` and the original data value is returned which was the previous behavior, in your case we don't return the original input anymore if `iRet is not >=0`. I tried to be as unobtrusive and preserve the original behavior here as I don't want to change some very old behavior in our code base, just to fix this warning.